httpx/client.py

```diff
@@ -176,6 +179,13 @@ def merge_cookies(
             response = await self.dispatch.send(
                 request, verify=verify, cert=cert, timeout=timeout
             )
+            # TODO: requests also checks for number of 401 calls < 2
+            if response.is_client_error and response.expects_digest_auth:
```

Comment:
We're going to need a better way to allow users to use this behavior branch more generally. Realm-based response-aware authentication class that only triggers when certain status codes / headers are available?